snuba/snuba_migrations/functions/0004_functions_v2.py

```diff
@@ -0,0 +1,197 @@
+        AggregateFunction("quantiles(0.5, 0.75, 0.9, 0.95, 0.99)", [Float(64)]),
+    ),
+    Column("min", AggregateFunction("min", [Float(64)])),
+    Column("max", AggregateFunction("max", [Float(64)])),
+    Column("avg", AggregateFunction("avg", [Float(64)])),
+    Column("sum", AggregateFunction("sum", [Float(64)])),
+    Column(
+        "worst",
+        AggregateFunction(
+            "argMax",
+            [
+                Tuple(
+                    (
+                        UUID(),  # profile_id
+                        String(),  # thread_id
+                        DateTime64(  # start_timestamp
+                            precision=6,
+                            modifiers=Modifiers(nullable=True),
+                        ),
+                        DateTime64(  # end_timestamp
+                            precision=6,
+                            modifiers=Modifiers(nullable=True),
+                        ),
+                    )
+                ),  # end Tuple
+                Float(64),
+            ],
+        ),
+    ),
+    Column(
+        "examples",
+        AggregateFunction(
+            "groupUniqArray(5)",
+            [
+                Tuple(
+                    (
+                        UUID(),  # profile_id
+                        String(),  # thread_id
+                        DateTime64(
+                            precision=6,
+                            modifiers=Modifiers(nullable=True),
+                        ),  # start_timestamp
+                        DateTime64(
+                            precision=6,
+                            modifiers=Modifiers(nullable=True),
+                        ),  # end_timestamp
+                    )
+                ),  # end Tuple
+            ],
+        ),
+    ),
+    Column("retention_days", UInt(16)),
+]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+    storage_set = StorageSetKey.FUNCTIONS
+
+    data_granularity = 60 * 60  # 1 hour buckets
+
+    local_raw_table = "functions_raw_local"
+
+    local_functions_table = "functions_v2_local"
+    dist_functions_table = "functions_v2_dist"
+
+    local_mv_table = "functions_mv_v2_local"
+
+    def forwards_ops(self) -> Sequence[SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=self.storage_set,
+                table_name=self.local_functions_table,
+                columns=columns,
+                engine=table_engines.AggregatingMergeTree(
+                    storage_set=self.storage_set,
+                    order_by="(project_id, timestamp, transaction_name, fingerprint, name, package, is_application, profiling_type, platform, environment, release, retention_days)",
+                    primary_key="(project_id, timestamp, transaction_name, fingerprint)",
+                    partition_by="(retention_days, toMonday(timestamp))",
+                    settings={"index_granularity": "2048", "allow_nullable_key": 1},
```

Comment:
@Zylphrex I've carried over this specific `index_granularity` settings from the previous `functions` mv.

Can you remind me why this was picked specifically instead of the default granule size of `8012`? Did you ran some experiments on existing functions data and it turned out the queries were too slow with `8012`?

Reply:
I'm not sure, it may have just a copy paste from another table.